query/src/main/java/org/infinispan/query/impl/massindex/NonFlushingIndexWorker.java

```diff
@@ -0,0 +1,58 @@
+package org.infinispan.query.impl.massindex;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.query.impl.externalizers.ExternalizerIds;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+/**
+ * Index worker that does not purge or flush the index.
+ *
+ * @author gustavonalle
+ * @since 7.1
+ */
+public final class NonFlushingIndexWorker extends AbstractIndexWorker {
+
+   public NonFlushingIndexWorker(Class<?> entity, String indexName) {
+      super(entity, indexName);
+   }
+
+   @Override
+   protected void preIndex() {
```

Comment:
The difference between flushing/non-flushing is so small that a simple boolean flag can solve it :) Is it worth having a hierarchy of 3 classes and two identical externalizers?